controller/environment.go

```diff
@@ -132,3 +143,17 @@ func (c *EnvironmentController) Show(ctx *app.ShowEnvironmentContext) error {
 	}
 	return ctx.OK(res)
 }
+
+func (c *EnvironmentController) checkClustersUser(ctx context.Context, clusterURL string) error {
+	clusters, err := c.clusterService.UserClusters(ctx)
+	if err != nil {
+		return err
+	}
+	clusterURL = httpsupport.RemoveTrailingSlashFromURL(clusterURL)
+	for _, cluster := range clusters.Data {
+		if httpsupport.RemoveTrailingSlashFromURL(cluster.APIURL) == clusterURL {
+			return nil
+		}
+	}
+	return errors.NewInternalErrorFromString(fmt.Sprintf("cluster with URL '%s' not linked with user account", clusterURL))
```

Comment:
why do you return an `InternalServerError` here? Could you return something more explicit, such as `Forbidden` if the cluster is not linked to the user?